src/mkdocstrings/extension.py

```diff
@@ -132,7 +135,12 @@ def run(self, parent: Element, blocks: Element) -> None:
             handler.renderer.update_env(self.md, self._config)
 
             log.debug("mkdocstrings.extension: Rendering templates")
-            rendered = handler.renderer.render(data, rendering)
+            try:
+                rendered = handler.renderer.render(data, rendering)
+            except TemplateNotFound as error:
+                log.error(f"mkdocstrings.extension: Template \"{error.name}\" not found "
+                          f"for \"{handler.renderer.directory}\" handler and theme \"{self._config['theme_name']}\".")
```

Comment:
You can use single quotes `'` instead of escaped double-quotes `\"`.
Also you can assign `theme_name = self._config["theme_name"]` before to simplify the f-string :slightly_smiling_face:

Reply:
The user doesn't really need to know what directory the handler is using. You could just use `handler_name` instead of `handler.renderer.directory`, and avoid assigning `directory` to `self.directory` in the base renderer's constructor.

Reply:
Yes, I was looking for a name and I did see that `handler_name` was available. This is updated in 
e275221